### PR TITLE
Updated pickers with input attribute compatibility.

### LINKED ISSUE #22528: NormalPeoplePicker required attribute
#### Describe the feature that you would like added

I would like to have a "required" and "onGetErrorMessage" attributes be added

#### What component or utility would this be added to

NormalPeoplePicker (ideally it would be on the BasePicker so all pickers would have it)

#### Have you discussed this feature with our team, and if so, who

No

#### Additional context/screenshots

The expected behavior would be similar to the same attributes of the ComboBox.

### PR DESCRIPTION
**Pull request by tpalacino**
## Previous Behavior

The Picker components that extend the `BasePicker` and `BasePickerListBelow` are missing properties that most other FluentUI user input component support. The missing properties include **label**, **required**, **errorMessage**, and **onGetErrorMessage**.

## New Behavior

The pickers now support the following optional properties:
| Property | Description |
| --------- | ------------- |
| **label** | When specified renders consistently with how a label is rendered for a `TextField`.|
| **required** | When specified renders consistently with how the required attribute is rendered for a `TextField`.|
| **errorMessage** | When specified renders consistently with how an error message is rendered for a `ComboBox`.|
| **onGetErrorMessage** | When specified behaves consistently with how an error message function behaves for a `TextField`.|

## Related Issue(s)

- Fixes #22528

### CODE REVIEW

**Comment by tpalacino** [comment]
@gouttierre @msft-fluent-ui-bot @JustSlone @layershifter @khmakoto @micahgodbolt Can someone please help me merge this?

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (smhigley) [inline packages/react/src/components/pickers/BasePicker.tsx:368]: This would need to be associated with the input through `aria-describedby` when it is available, without overriding or replacing the other `aria-describedby` value. The `aria-describedby` attribute can take a space-separated list of ids that are concatenated in order as the description.

Textfield does this by inserting the error message in the same element as the static description, so sharing the same parent `id` association.

File: packages/react/src/components/pickers/BasePicker.types.ts
Comment (smhigley) [inline packages/react/src/components/pickers/BasePicker.types.ts:55]: I get why you added this because Textfield does have it, but we now see it as an antipattern to add custom aliases for ARIA props. I think it'd be good to leave it off new additions to avoid adding more instances of this sort of attribute. Since we already support the native `aria-label`, I don't think BasePicker needs to also add `ariaLabel`.

Let me know if I'm missing a use case for this though!

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (smhigley) [inline packages/react/src/components/pickers/BasePicker.tsx:386]: do we also need to add a way to pass styles through to the label, similar to Textfield's `subComponentStyles`?

I also don't believe the `Label` control needs an `id` here, and I don't see it used anywhere else in the added code. Let me know if I'm missing somewhere it's being used though.

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (smhigley) [inline packages/react/src/components/pickers/BasePicker.tsx:1057]: We generally do not include strings in our components, since we don't handle localization. I think the default here would be to rely on `errorMessage` and `onGetErrorMessage` and not perform validation or render a default message.

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (smhigley) [inline packages/react/src/components/pickers/BasePicker.tsx:191]: I think the check would need to be whether `onGetErrorMessage` returns a string, since there could be other types of validation based on selected items & value beyond the required prop.

**Review by smhigley — CHANGES_REQUESTED** [review]
Thanks for doing this work! I did an initial pass and left comments on stuff I saw. Let me know if they make sense, and if you have any questions!

File: packages/react-examples/src/react/PeoplePicker/PeoplePicker.List.Example.tsx
Comment (khmakoto) [inline packages/react-examples/src/react/PeoplePicker/PeoplePicker.List.Example.tsx:204]: In case you go with the suggestion above

```suggestion
          onChange={onPickerLabelChange}
```

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (khmakoto) [inline packages/react/src/components/pickers/BasePicker.tsx:307]: If you're not passing anything else here, you can just do:

```suggestion
          error: 'ms-BasePicker-error',
          label: 'ms-BasePicker-label',
```

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (khmakoto) [inline packages/react/src/components/pickers/BasePicker.tsx:315]: _nit:_

```suggestion
    const inputId = inputProps?.id ?? this._ariaMap.combobox;
```

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (khmakoto) [inline packages/react/src/components/pickers/BasePicker.tsx:387]: _nit:_ You could also achieve this, possibly even with better performance, by doing simple string concatenation instead of pushing into an array and joining.

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (khmakoto) [inline packages/react/src/components/pickers/BasePicker.tsx:285]: I think it's probably better if you define this as you did below:

```suggestion
    const hasError = !!(this.state.errorMessage ?? this.props.errorMessage);
```

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (khmakoto) [inline packages/react/src/components/pickers/BasePicker.tsx:408]: Should these be the other way around?

```suggestion
    const errorMessage = this.state.errorMessage ?? this.props.errorMessage;
```

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (khmakoto) [inline packages/react/src/components/pickers/BasePicker.tsx:1204]:
```suggestion
    const hasError = !!(this.state.errorMessage ?? this.props.errorMessage);
```

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (khmakoto) [inline packages/react/src/components/pickers/BasePicker.tsx:1226]: Same thing as above, you don't need to use `css` here if you are only passing one classname

```suggestion
          error: 'ms-BasePicker-error',
          label: 'ms-BasePicker-label',
```

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (khmakoto) [inline packages/react/src/components/pickers/BasePicker.tsx:1239]: _nit:_

```suggestion
    const inputId = inputProps?.id ?? this._ariaMap.combobox;
```

File: packages/react/src/components/pickers/BasePicker.tsx
Comment (tpalacino) [reply packages/react/src/components/pickers/BasePicker.tsx:408]: I copied the implementation from the `_errorMessage` property of the TextField.base.tsx

**Review by khmakoto — APPROVED** [review]

**Review by smhigley — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

**Comment by tpalacino** [comment]
@khmakoto What are the next steps?

**Comment by khmakoto** [comment]
@tpalacino I'll merge your PR once all checks have completed (I just approved the workflows so CI starts to run those checks).

**Pull request merged**
